src/server/game/Entities/Object/Updates/ViewerDependentValues.h

```diff
@@ -146,6 +147,30 @@ class ViewerDependentValue<UF::UnitData::DisplayIDTag>
     }
 };
 
+template<>
+class ViewerDependentValue<UF::ObjectData::EntryIDTag>
+{
+public:
+    using value_type = UF::ObjectData::EntryIDTag::value_type;
+
+    static value_type GetValue(UF::ObjectData const* objectData, Object const* object, Player const* receiver)
+    {
+        value_type entryID = objectData->EntryID;
+
+        if (Unit const* unit = object->ToUnit())
+        {
+            if (receiver == unit->ToTempSummon()->GetSummoner())
```

Comment:
this obviously wasnt tested ingame because this line will immediately crash the server on first visibility update for first logging in player

Reply:
it's draft, not tested yet, gimme some time to finish it up, I got the basics done (mostly) 😿

Reply:
This should be fixed by checking if the creature is a temporal summon now.